src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,113 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/common/prisma/prisma.service';
+import * as bcrypt from 'bcryptjs';
+import { Response } from 'express';
+import { JwtService } from '@nestjs/jwt';
+import { CreateUserDto } from './dto/create-user.dto';
+import { TokenPair } from './interfaces/token-pair.interface';
+import { User } from '../user/interfaces/user.interface';
+import { AuthResponse } from './interfaces/auth-response.interface';
+import { UserService } from '../user/user.service';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prismaService: PrismaService,
+    private jwtService: JwtService,
+    private userService: UserService,
+  ) {}
+
+  async registration(
+    userData: CreateUserDto,
+    res: Response,
+  ): Promise<AuthResponse> {
+    const candidate = await this.userService.findUserByEmail(userData.email);
+    if (candidate) {
+      throw new HttpException(
```

Comment:
`throw new BadRequstException('User with this email already exists')`
It is better